### PR TITLE
Add unit test for registered property descriptors and fix discrepancies

### LINKED ISSUE #1872: Invalid Console warnings on boolean attributes
**Describe the issue**
The consoles shows warnings that are most probably not correct: 
CsvParserPipe [csvToXml] does not have an attribute [fileContainsHeader] to set to value [true]

There is an attribute fileContainsHeader in the class CsvParserPipe.

And probably more console warnings on other boolean attributes.

**Reporter**
Richard van der Stel

**To Reproduce**

**Screenshots**
image

**Configuration**

            <pipe name="csvToXml"
                  className="nl.nn.adapterframework.pipes.CsvParserPipe"
                  fileContainsHeader="true"
                  fieldSeparator=";"
            >
                <forward name="exception"   path="fail:selectNumberOfQueryRecords"/>
                <forward name="success"     path="processBlocks"/>
            </pipe>

**Input**
```
Please provide an example of the input message. 
Alternatively, provide a Ladybug report.
```

**Environment**
FF! 7.7-20210527.152350: Ibis4Titan
Running on LPZWNL000022420 using Apache Tomcat/8.5.34
Java Version: Java(TM) SE Runtime Environment (1.8.0_60-b27)
Heap size: 1150M, total JVM memory: 1534M
Free disk space: 12GB, total disk space: 236GB
Up since: 2021-06-01 14:08:25 (20m)

**Additional Environment**
 - DBMS: [e.g. Oracle, MSSQL, MariaDB, MySql, PostgreSQL]
 - Browser: [e.g. Chrome, Safari, Edge, Firefox]

**Additional Context**
Add any other context about the problem here. (f.e. ladybug report / test adapter with larva test)

### PR DESCRIPTION
**Pull request by alisihab**
Closes #1872

### CODE REVIEW

**Review by gvanbrakel — CHANGES_REQUESTED** [review]

File: core/src/test/java/nl/nn/adapterframework/configuration/digester/MapPropertyDescriptorsTest.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+					String getterStartsWithIs = "is" + propertyName;
+
+					propertyName = propertyName.substring(0, 1).toLowerCase() + propertyName.substring(1);
+
+					boolean getterMatches = Arrays.stream(methodDescriptors)
+							.anyMatch(name -> name.getName().equals(getterName) || name.getName().equals(getterStartsWithIs));
+
+					if(getterMatches) {
+						PropertyDescriptor[] pds = beanInfo.getPropertyDescriptors();
+						PropertyDescriptor pd = Arrays.stream(pds)
```
Comment (nielsm5) [inline core/src/test/java/nl/nn/adapterframework/configuration/digester/MapPropertyDescriptorsTest.java:65]: Can't you just do `PropertyUtils.getWriteMethod(pd);` ?

File: core/src/test/java/nl/nn/adapterframework/configuration/digester/MapPropertyDescriptorsTest.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+			// get methods 
+			MethodDescriptor[] methodDescriptors =  beanInfo.getMethodDescriptors();
+			for (MethodDescriptor methodDescriptor : methodDescriptors) {
+				String methodName = methodDescriptor.getName();
+				if(methodName.startsWith("set")) {
+					String propertyName = methodName.substring(3);
+					String getterName = "get" + propertyName;
+					String getterStartsWithIs = "is" + propertyName;
+
+					propertyName = propertyName.substring(0, 1).toLowerCase() + propertyName.substring(1);
```
Comment (nielsm5) [inline core/src/test/java/nl/nn/adapterframework/configuration/digester/MapPropertyDescriptorsTest.java:58]: Perhaps `PropertyUtils.getPropertyDescriptor(bean, method);` might be easier here?

File: core/src/main/java/nl/nn/adapterframework/pipes/CsvParserPipe.java [hunk 3]
```diff
@@ -93,7 +93,7 @@ public PipeRunResult doPipe(Message message, PipeLineSession session) throws Pip
 	}
 
 	@IbisDoc({"1", "Specifies if the first line should be treated as header or as data", "true"})
```
[the remark this answers is not in the document]
Comment (alisihab) [reply core/src/main/java/nl/nn/adapterframework/pipes/CsvParserPipe.java:95]: I'll check this in the other PR

[the remark this answers is not in the document]
Comment (alisihab) [reply core/src/main/java/nl/nn/adapterframework/pipes/CsvParserPipe.java:95]: better to do it here to prevent any conflict

**Comment by alisihab** [comment]
merged to #1889

**Pull request closed without merging**